src/test/java/org/elasticsearch/test/ElasticsearchIntegrationTest.java

```diff
@@ -855,8 +855,14 @@ public void waitNoPendingTasksOnAll() throws Exception {
             @Override
             public void run() {
                 for (Client client : clients()) {
+                    ClusterHealthResponse clusterHealth = client.admin().cluster().prepareHealth().setLocal(true).get();
+                    assertThat("client " + client  + " still has fetch shard started", clusterHealth.getNumberOfFetchShardStarted(), equalTo(0));
+                    assertThat("client " + client  + " still has fetch shard store", clusterHealth.getNumberOfFetchShardStore(), equalTo(0));
                     PendingClusterTasksResponse pendingTasks = client.admin().cluster().preparePendingClusterTasks().setLocal(true).get();
                     assertThat("client " + client + " still has pending tasks " + pendingTasks.prettyPrint(), pendingTasks, Matchers.emptyIterable());
+                    clusterHealth = client.admin().cluster().prepareHealth().setLocal(true).get();
```

Comment:
I kept it as is, to keep the assert message above listing all the pending tasks, compared to just giving a number